src/components/modal/modal.tsx

```diff
@@ -36,7 +49,12 @@ export const Modal: FC<ModalProps> = ({
 
     return (
         <Transition appear show={isOpen} as={Fragment}>
-            <Dialog initialFocus={initialFocusInputRef} onClose={onActionSecondary}>
+            <Dialog
+                initialFocus={
+                    initial === InitialElement.INPUT ? initialFocusInputRef : initalFocusButtonRef
+                }
+                onClose={onActionSecondary}
```

Comment:
maybe rename those
```
onActionPrimary,
onActionSecondary,
buttonLabelPrimary,
buttonLabelSecondary
```
to 
`onSubmit, onCancel, labelSubmit, labelCancel` instead